app/Models/Mship/Account.php

```diff
@@ -365,6 +420,16 @@ public function addNote($noteType, $noteContent, $writer = null, $attachment = n
         return $note;
     }
 
+    public function roster(): HasOne
+    {
+        return $this->hasOne(Roster::class, 'account_id', 'id');
+    }
+
+    public function onRoster(): bool
+    {
+        return $this->roster?->account_id == $this->id;
```

Comment:
Can you just query the relationships existence instead of a null attribute check?

Reply:
so I originally had this as `$this->roster->exists()` but that actually causes a query to be issued. Could do a `!== null`

Reply:
Fine to leave as is then!